sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -371,7 +371,8 @@ object CreateStruct {
       // We should always use the last part of the column name (`c` in the above example) as the
       // alias name inside CreateNamedStruct.
       case (u: UnresolvedAttribute, _) => Seq(Literal(u.nameParts.last), u)
-      case (u @ UnresolvedExtractValue(_, e: Literal), _) => Seq(e, u)
+      // SPARK-43522: make sure the name always be a string literal.
+      case (u @ UnresolvedExtractValue(_, e: Literal), _) => Seq(Literal(e.value.toString), u)
```

Comment:
I think this can match map kep lookup as well, and it's unclear what `e.value.toString` will return if the literal data type is non-trivial, such as array and map.

How about we simply add `if e.dataType == StringType`? Then for your case it will fall back to the default name `s"col${index + 1}"`.

Reply:
Thanks for remind! This solution would be better.